### PR TITLE
Add an `all` source to the social dominance metric

### PR DESCRIPTION
**Pull request by valo**
The new source will fetch the social dominance across all sources and
average them.

### CODE REVIEW

File: lib/sanbase/social_data/social_dominance.ex [hunk 1]
```diff
@@ -11,6 +11,42 @@ defmodule Sanbase.SocialData.SocialDominance do
...
+        result
+      end)
+      |> List.zip()
+      |> Enum.map(&Tuple.to_list/1)
+      |> Enum.map(fn all_sources_point ->
+        average_dominance =
+          all_sources_point
+          |> Enum.map(fn point -> point.dominance end)
+          |> Enum.sum()
+          |> Kernel./(length(@sources))
```
Comment (IvanIvanoff) [inline lib/sanbase/social_data/social_dominance.ex]: Shouldn't there be some weight depending on the volume?

File: lib/sanbase_web/graphql/schema/social_data_queries.ex [hunk 2]
```diff
@@ -21,13 +21,14 @@ defmodule SanbaseWeb.Graphql.Schema.SocialDataQueries do
         2. "TELEGRAM" - shows the relative social dominance of this project in the telegram crypto channels
         3. "DISCORD" - shows the relative social dominance of this project on discord crypto communities
         4. "REDDIT" - shows the relative social dominance of this project on crypto subreddits
+        5. "ALL" - shows the average value of the social dominance across all sources
```
Comment (IvanIvanoff) [inline lib/sanbase_web/graphql/schema/social_data_queries.ex]: nitpick: here in the docs if we quote the params they look like string but in reality they are used without quotes

File: lib/sanbase/social_data/social_dominance.ex (hunk 1 above)
Comment (valo) [reply lib/sanbase/social_data/social_dominance.ex]: Let's roll it out like this and do fixes as we find issues. As of right now we don't have an idea if volume needs to be taken into account or not.

File: test/sanbase/social_data/social_dominance_test.exs [hunk 3]
```diff
@@ -58,6 +59,44 @@ defmodule Sanbase.SocialDominanceTest do
...
+    test "when computing for all sources" do
+      from = DateTime.from_naive!(~N[2018-04-16 10:00:00], "Etc/UTC")
+      to = DateTime.from_naive!(~N[2018-04-16 22:00:00], "Etc/UTC")
+
+      with_mock(HTTPoison, [],
+        get: fn _, _, _ ->
+          {:ok,
+           %HTTPoison.Response{
+             body:
+               "[{\"BTC_bitcoin\": 5, \"EOS_eos\": 15, \"ETH_ethereum\": 5, \"datetime\": 1523872800}, {\"BTC_bitcoin\": 15, \"EOS_eos\": 5, \"ETH_ethereum\": 10, \"datetime\": 1523916000}]",
```
Comment (kalkov) [inline test/sanbase/social_data/social_dominance_test.exs]: Will be more readable on multiple lines.

File: lib/sanbase_web/graphql/schema/social_data_queries.ex (hunk 2 above)
Comment (valo) [reply lib/sanbase_web/graphql/schema/social_data_queries.ex]: True that. I'll remove the quotes

**Pull request merged**